core/src/test/scala/org/apache/spark/deploy/StandaloneDynamicAllocationSuite.scala

```diff
@@ -388,25 +388,23 @@ class StandaloneDynamicAllocationSuite
     val executors = getExecutorIds(sc)
     val executorIdsBefore = executors.toSet
     assert(executors.size === 2)
-    // kill executor 1, and replace it
+    // kill and replace an executor
     assert(sc.killAndReplaceExecutor(executors.head))
     eventually(timeout(10.seconds), interval(10.millis)) {
       val apps = getApplications()
       assert(apps.head.executors.size === 2)
       val executorIdsAfter = getExecutorIds(sc).toSet
-      // make sure the old executors head has been killedAndReplaced.
+      // make sure the executor was killed and replaced
       assert(executorIdsBefore != executorIdsAfter)
     }
 
-    var apps = getApplications()
-    // kill executor 1, and actually nothing to kill
-    assert(!sc.killExecutor(executors.head))
```

Comment:
We don't have to but it doesn't hurt, you can add something like
```
val executorToReplace = executors.head
assert(sc.killAndReplaceExecutor(executorToReplace))
...

// cannot kill the same executor again because it was already killed
assert(!sc.killExecutor(executorToReplace))
```